snap/snapcraft.yaml

```diff
@@ -0,0 +1,49 @@
+name: charmed-openstack-exporter
+base: core22
+summary: OpenStack Exporter for Prometheus
+description: |
+  The OpenStack exporter, exports Prometheus metrics from a running OpenStack
+  cloud for consumption by prometheus.
+adopt-info: openstack-exporter
+confinement: strict
+grade: stable
+
+apps:
+  openstack-exporter:
+    command: bin/openstack-exporter
+    plugs: [home, network, network-bind]
+  service:
+    command: bin/service
+    daemon: simple
+    install-mode: disable
+    restart-condition: on-abnormal
+    plugs:
+      - home
+      - network
+      - network-bind
+      - etc-openstack
+
+parts:
+  openstack-exporter:
+    plugin: go
+    source: https://github.com/openstack-exporter/openstack-exporter.git
+    source-type: git
+    build-snaps:
+      - go
+    # override pull to read VERSION because working dir is root of source repo
+    override-pull: |
+      craftctl default
+      craftctl set version="$(cat VERSION)"
```

Comment:
These need to be removed because the `VERSION` file is only available in the upstream project. Or it can be improved e.g.

```
    override-pull: |
      craftctl default
      TAG=$(git describe --tags --abbrev=10)
      craftctl set version="${TAG#v}"
```
or

(this need annotated tags)
```
version: git
```

Reply:
Nice catch. It was working to cat the `VERSION` from the upstream, however it's more accurate to get from the git itself. E.g: on upstream is using `1.6.0`, but the current version is `1.7.x`. Probably the maintainer forgot to update the `VERSION` file. I think we should change also in upstream to be more realistic in the version, otherwise is very easy to forget when building the snap and releasing it.

Reply:
Agreed, if we make improvements to the snapcraft file, it would be nice to contribute upstream.

@chanchiwai-ray is referring to this upstream issue: https://github.com/openstack-exporter/openstack-exporter/issues/357 - this suggestion will fix that.  :)